src/keystore/sdcard.py

```diff
@@ -115,18 +164,61 @@ async def delete_mnemonic(self, path=None):
             if not platform.is_sd_present():
                 raise KeyStoreError("SD card is not present")
             platform.mount_sdcard()
-        if not platform.file_exists(path):
-            raise KeyStoreError("Secret is not saved. No need to delete anything.")
+
+        file = await self.select_file(path, self.fileprefix(path))
+        if file is None:
+            return False
+
+        if not platform.file_exists("%s/%s" % (path, file)):
+            raise KeyStoreError("File not found.")
         try:
-            os.remove(path)
+            os.remove("%s/%s" % (path, file))
         except Exception as e:
             print(e)
-            raise KeyStoreError("Failed to delete file at " + path)
+            raise KeyStoreError("Failed to delete file '%s' from %s" % (file, path))
         finally:
             if path == self.sdpath:
                 platform.unmount_sdcard()
             return True
 
+    async def get_input(
+            self,
+            title="Enter a name for this seed",
+            note="Naming your seeds allows you to store multiple.\n"
+                 "Give each seed a unique name!",
+            suggestion="",
+    ):
+        scr = InputScreen(title, note, suggestion)
+        await self.show(scr)
+        return scr.get_value()
+
+    async def export_mnemonic(self):
+        if await self.show(Prompt("Warning",
+                                  "You need to confirm your PIN code "
+                                  "to export your recovery phrase.\n\n"
+                                  "Continue?")):
+            self.lock()
+            await self.unlock()
+
+            seed = bip39.mnemonic_to_seed(self.mnemonic)
+            filename = "seed-export-%s.txt" % hexlify(sha256(seed).digest()[:4])
```

Comment:
Maybe first word instead of the fingerprint? Then we don't need to do expensive 2048 hashing rounds

```suggestion
            filename = "seed-export-%s.txt" % self.mnemonic.split()[0]
```